elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -431,6 +435,28 @@ private JsonObject getTenantIdFilter() {
 			}
 		}
 
+		private void addFacetingRequest(JsonBuilder.Object facets, FacetingRequest facetingRequest) {
+			String facetFieldAbsoluteName = facetingRequest.getFieldName();
+			FacetMetadata facetMetadata = null;
+			for ( Class<?> entityType : getQueriedEntityTypes() ) {
+				EntityIndexBinding binding = extendedIntegrator.getIndexBinding( entityType );
+				TypeMetadata typeMetadata = binding.getDocumentBuilder().getTypeMetadata();
+				facetMetadata = typeMetadata.getFacetMetadataFor( facetFieldAbsoluteName );
+				if ( facetMetadata != null ) {
+					break;
+				}
+			}
+
+			if ( facetMetadata == null ) {
+				throw new AssertionFailure( "Unknown facet field name:" + facetFieldAbsoluteName );
```

Comment:
Didn't look at the path to arrive here but we are sure the user can't get this error? If he can, we should move this the logger.